hail_search/queries/mito.py

```diff
@@ -112,6 +118,215 @@ class MitoHailTableQuery(BaseHailTableQuery):
+    ):
+        filtered_hts = []
+        filtered_comp_het_hts = []
+        wes_entries, wgs_entries = entries_hts_map[SAMPLE_TYPE_WES], entries_hts_map[SAMPLE_TYPE_WGS]  # entries_hts_map: {<sample_type>: [(ht, project_samples), ...]}
+
+        # Iterate over each chunk of project tables
+        for (wes_ht, wes_project_samples), (wgs_ht, wgs_project_samples) in zip(wes_entries, wgs_entries):
+            wes_ht, sorted_wes_family_sample_data = self._add_entry_sample_families(wes_ht, wes_project_samples, SAMPLE_TYPE_WES)
+            wgs_ht, sorted_wgs_family_sample_data = self._add_entry_sample_families(wgs_ht, wgs_project_samples, SAMPLE_TYPE_WGS)
+            family_idx_map = self._build_family_index_map(sorted_wes_family_sample_data, sorted_wgs_family_sample_data)
+
+            wes_ht = wes_ht.rename({'family_entries': WES_FAMILY_ENTRIES_FIELD, 'filters': WES_FILTERS_FIELD})
+            wgs_ht = wgs_ht.rename({'family_entries': WGS_FAMILY_ENTRIES_FIELD, 'filters': WGS_FILTERS_FIELD})
+            ht = wes_ht.join(wgs_ht, how='outer')
+
+            # Filter by quality
+            ht = self._annotate_passes_quality_filter(ht, quality_filter, **kwargs)
+
+            # Filter by inheritance
+            any_valid_entry, is_any_affected = self._get_any_family_member_gt_has_alt_filter()
+            ht = ht.annotate(wes_passes_inheritance=ht[WES_FAMILY_ENTRIES_FIELD].map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ))
+            ht = ht.annotate(wgs_passes_inheritance=ht[WGS_FAMILY_ENTRIES_FIELD].map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ))
+
+            if self._has_comp_het_search:
+                comp_het_ht = self._annotate_families_inheritance(ht, COMPOUND_HET, inheritance_filter, sorted_wes_family_sample_data, SAMPLE_TYPE_WES)
+                comp_het_ht = self._annotate_families_inheritance(comp_het_ht, COMPOUND_HET, inheritance_filter, sorted_wgs_family_sample_data, SAMPLE_TYPE_WGS)
+                comp_het_ht = self._apply_filters(comp_het_ht, family_idx_map)
+                filtered_comp_het_hts.append(comp_het_ht)
+
+            if is_any_affected or not (inheritance_filter or self._inheritance_mode):
+                # No sample-specific inheritance filtering needed
+                sorted_wes_family_sample_data = sorted_wgs_family_sample_data = []
+
+            if self._inheritance_mode != COMPOUND_HET:
+                ht = self._annotate_families_inheritance(ht, self._inheritance_mode, inheritance_filter, sorted_wes_family_sample_data, SAMPLE_TYPE_WES)
+                ht = self._annotate_families_inheritance(ht, self._inheritance_mode, inheritance_filter, sorted_wgs_family_sample_data, SAMPLE_TYPE_WGS)
+                ht = self._apply_filters(ht, family_idx_map)
+                filtered_hts.append(ht)
+
+        merged_ht = self._merge_project_hts(filtered_hts, n_partitions)
```

Comment:
I don't think it makes sense to change the overall flow for how multiple project hts are merged as part of this PR - if at the end of the day you are generating each project ht with the correct sample types and filtering, why would the logic of combining multiple projects need to change to support this functionality?

Reply:
Ah for some reason I thought that the original logic would not ensure that the same project would be in each chunk across both sample types, but it actually does and you're right I don't need to change that logic.